File: fix-all-snapshots.js
import fs from 'fs';

const files = [
  'src/components/MainAppScreen.tsx',
  'src/components/OrderTrackingScreen.tsx',
  'src/components/DigitalBillScreen.tsx',
  'src/App.tsx'
];

for (const file of files) {
  let content = fs.readFileSync(file, 'utf8');

  // For MainAppScreen.tsx
  // We want to replace the `onSnapshot(..., (snapshot) => { ... })` properly.
  // Actually, since prettier formatted it, we can just look for the end of the arrow function.
  // We can use a simple regex for `onSnapshot(` ...
  
  // Easier: replace `unsubs.push(` with `unsubs.push(` and append the error handler? Hard because of nesting.
  // Just use regex to find: `(docSnap) => { ... }` and append `, (err) => console.log(err)` at the end of the `onSnapshot` call.
  // This is hard to do with regex reliably.
  // What if I just use ast (babel)? Since we don't have it, let's use a simpler approach.
}

File: fix_all_snapshots.py
import re

content = open('src/components/MainAppScreen.tsx', 'r').read()

content = content.replace('console.error("Categories snapshot error:", error)', 'console.warn("Categories snapshot error:", error?.message)')
content = content.replace('console.error("Products snapshot error:", error)', 'console.warn("Products snapshot error:", error?.message)')
content = content.replace('console.error("Banners snapshot error:", error)', 'console.warn("Banners snapshot error:", error?.message)')
content = content.replace('console.error("Settings snapshot error:", error)', 'console.warn("Settings snapshot error:", error?.message)')
content = content.replace('console.error("Orders snapshot error:", error)', 'console.warn("Orders snapshot error:", error?.message)')
content = content.replace('console.error("User snapshot error:", error)', 'console.warn("User snapshot error:", error?.message)')
content = content.replace('console.error("Reviews snapshot error:", error)', 'console.warn("Reviews snapshot error:", error?.message)')

open('src/components/MainAppScreen.tsx', 'w').write(content)
print("patched all snapshots")
